src/System.Private.DataContractSerialization/src/System/Xml/XmlUTF8TextReader.cs

```diff
@@ -892,6 +892,24 @@ private void ReadAttributes()
             ProcessAttributes();
         }
 
+        // NOTE: Call only if 0xEF has been seen in the stream AND there are three valid bytes to check (buffer[offset], buffer[offset + 1], buffer[offset + 2]). 
+        // 0xFFFE and 0xFFFF are not valid characters per Unicode specification. The first byte in the UTF8 representation is 0xEF. 
+        bool IsNextCharacterNonFFFE(byte[] buffer, int offset)
```

Comment:
Afaik, as convention we specify method scope keywords explicitly for method so we should use ```private``` here. Similar comments for the newly added methods.